core/node/core.go

```diff
@@ -82,6 +84,13 @@ func (s *syncDagService) Session(ctx context.Context) format.NodeGetter {
 	return merkledag.NewSession(ctx, s.DAGService)
 }
 
+// Resolver returns a resolver that's configured to look up unixfs paths
+func Resolver(bs blockservice.BlockService) *resolver.Resolver {
+	rs := resolver.NewBasicResolver(bs)
+	rs.FetchConfig.NodeReifier = unixfsnode.Reify
```

Comment:
I agree. I think we should have a FetchConfig top level dependency in the DI stack, and pass it to NewBasicResolver rather than the block service. I think I've said this elsewhere as well.